meteor/scale.py

```diff
@@ -1,15 +1,222 @@
+            + np.square(to_scale_uncertainties.loc[common_miller_indices])
+        )
+    else:
+        uncertainty_weights = 1.0
+
+    common_reference_values = reference_values.loc[common_miller_indices]
+    common_values_to_scale = values_to_scale.loc[common_miller_indices]
+
+    def compute_residuals(scaling_parameters: ScaleParameters) -> np.ndarray:
+        scale_factors = _compute_anisotropic_scale_factors(
+            common_miller_indices, scaling_parameters
+        )
+        return uncertainty_weights * (
+            scale_factors * common_values_to_scale - common_reference_values
+        )
+
+    initial_scaling_parameters: ScaleParameters = (1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0)
+    optimization_result = opt.least_squares(compute_residuals, initial_scaling_parameters)
+    optimized_parameters: ScaleParameters = optimization_result.x
+
+    # now be sure to compute the scale factors for all miller indices in `values_to_scale`
+    optimized_scale_factors = _compute_anisotropic_scale_factors(
+        values_to_scale.index, optimized_parameters
+    )
+    assert len(optimized_scale_factors) == len(values_to_scale)
+
+    return optimized_scale_factors
+
+
+def scale_datasets(
+    *,
+    reference_dataset: rs.DataSet,
+    dataset_to_scale: rs.DataSet,
+    column_to_compare: str = "F",
+    uncertainty_column: str = "SIGF",
+    weight_using_uncertainties: bool = True,
+) -> rs.DataSet:
+    """
+    Scale a dataset to align it with a reference dataset using anisotropic scaling.
+
+    This function scales the dataset (`dataset_to_scale`) by comparing it to a reference dataset
+    (`reference_dataset`) based on a specified column. The scaling applies an anisotropic model of
+    the form:
 
         C * exp{ -(h**2 B11 + k**2 B22 + l**2 B33 +
                     2hk B12 + 2hl  B13 +  2kl B23) }
 
-    This is the same procedure implemented by CCP4's SCALEIT.
+    The parameters Bxy are fit using least squares, optionally with uncertainty weighting.
+
+    NB! All intensity, amplitude, and standard deviation columns in `dataset_to_scale` will be
+    modified (scaled). To access the scale parameters directly, use
+    `meteor.scale.compute_scale_factors`.
+
+    Parameters:
+    -----------
+    reference_dataset : rs.DataSet
+        The reference dataset, containing a column with the values to which `dataset_to_scale` is
+        compared.
+    dataset_to_scale : rs.DataSet
+        The dataset to be scaled. It must contain the same columns as `reference_dataset`.
+    column_to_compare : str, optional (default: "F")
+        The column in both datasets that is used for the comparison and scaling.
+    uncertainty_column : str, optional (default: "SIGF")
+        The column in both datasets containing uncertainty values for weighting. Used only if
+        `weight_using_uncertainties` is True.
+    weight_using_uncertainties : bool, optional (default: True)
```

Comment:
I don't fully understand the implications of weight_using_uncertainties. The weights are applied to the scale factors, but they are not used in determining the scale factors themselves? I would have almost done the opposite